diagnostic_aggregator/src/aggregator.cpp

```diff
@@ -33,214 +33,285 @@
  *********************************************************************/
 
 /**! \author Kevin Watts */
+/**! \author Arne Nordmann */
 
-#include <diagnostic_aggregator/aggregator.h>
+#include "diagnostic_aggregator/aggregator.hpp"
 
-using namespace std;
-using namespace diagnostic_aggregator;
+#include <memory>
+#include <string>
+#include <vector>
 
-Aggregator::Aggregator() :
-  pub_rate_(1.0),
-  analyzer_group_(NULL),
-  other_analyzer_(NULL),
-  base_path_("")
+namespace diagnostic_aggregator
 {
-  ros::NodeHandle nh = ros::NodeHandle("~");
-  nh.param(string("base_path"), base_path_, string(""));
-  if (base_path_.size() > 0 && base_path_.find("/") != 0)
-    base_path_ = "/" + base_path_;
 
-  nh.param("pub_rate", pub_rate_, pub_rate_);
+using std::string;
+using std::vector;
+
+using rclcpp::get_logger;
+
+using std::placeholders::_1;
+using std::placeholders::_2;
+using std::placeholders::_3;
 
+/**
+ * @todo(anordman): make aggregator a lifecycle node.
+ */
+Aggregator::Aggregator()
+: n_(std::make_shared<rclcpp::Node>("analyzers",
```

Comment:
I believe this should be templated on Node. Ideally the `Aggregator` should be able to take a `rclcpp::Node` as well as a `rclcpp_lifecycle::Node`.

Reply:
I think we should rather go with `Aggregator` actually extending `rclcpp_lifecycle::Node`.

Reply:
This is a bit more tricky than I initally thought:

1. It can't be `rclcpp_lifecycle::LifecycleNode` easily, because the API of the `Bond` formed between the aggregator and the analyzers expects a `rclcpp:Node`. This might require refactoring of `Bond`.
2. The aggregator probably _has_ a node (as opposed to _being_ a node), because said `Bonds` get `shared_ptr` to the aggregator and analyzers. This would introduce a cyclic dependency since the aggregator node then would have `shared_ptrs` of the `Bonds`, and the `Bonds` would have `shared_ptrs` to the aggregator. This could maybe solved with the one of these pointers being `weak_ptrs`, but I'm not sure.